spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/core/PubSubConfiguration.java

```diff
@@ -27,13 +30,15 @@
  */
 public class PubSubConfiguration {
 
+	private final Map<String, PubSubProperties> properties = new HashMap<>();
```

Comment:
This is me being dense, because I think you already explained it, but what was the reason we could not have a direct `Map<String, PubSubConfiguration.Subscriber>`?

Reply:
When we were using `Map<String, PubSubConfiguration.Subscriber>` with `spring.cloud.gcp.pubsub.properties.[sub-name].subscriber.executor-threads` the subscription-specific properties weren't getting recognized and we were just falling back to the default Subscriber properties. So I followed this [example](https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config.typesafe-configuration-properties.merging-complex-types) to add an intermediate class. 
I haven't verified this design yet but `Map<String, PubSubConfiguration.Subscriber>` might work if our configuration setup looks like `spring.cloud.gcp.pubsub.properties.[sub-name].executor-threads` -- we would have to rename `properties` to something more descriptive such as `subscriber-properties`. What do you think?

Reply:
Give it a try to see if it works. If it does not, let's stick with the wrapper class. But if it does, I think `spring.cloud.gcp.pubsub.subscription.[sub-name].executor-threads` would be a bit more user friendly than `spring.cloud.gcp.pubsub.properties.[sub-name].subscriber.executor-threads`.

Reply:
It worked! The new setup looks much better.

Reply:
Oh, good. I was worried I was being a thorn in your side for no tangible benefit.